src/cluster.c

```diff
@@ -1820,6 +1862,102 @@ void clusterUpdateSlotsConfigWith(clusterNode *sender, uint64_t senderConfigEpoc
     }
 }
 
+/* Cluster ping extensions.
+ *
+ * The ping/pong/meet messages support arbitrary extensions to add additional
+ * metadata to the messages that are sent between the various nodes in the
+ * cluster. The extensions take the form:
+ * [ Header length + type (8 bytes) ] 
+ * [ Extension information (Arbitrary length, but must be 8 byte padded) ]
+ */
+
+
+/* Returns the length of a given extension */
+static uint16_t getPingExtLength(clusterMsgPingExt *ext) {
+    return ntohl(ext->length);
+}
+
+/* Returns true if the provided extension starts at the end of 
+ * the cluster message, indicating we are done parsing extensions */
+static int isTerminalPingExt(clusterMsgPingExt *ext, clusterMsg *hdr) {
```

Comment:
Not thrilled about relying on `hdr->totlen` to determine that we've reached the last ping ext, it leaves very little room to future changes.

Maybe `clusterMsgData.ping` could explicitly state the number of extensions included, or specify a `ping_totlen` for that specific payload?

Reply:
(: I did have an earlier check for the number of extensions and you asked if was needed. I can add it back, I think it was a better sanity check.

Reply:
Shame on me :)
On the first review, I had a misconception of the cluster bus extension mechanism when and considered it to be completely decoupled from specific message types.

Reply:
Ok, back to having the total number of extensions. We still validate it based off the length as well, but we could obviously change that implementation in the future.